core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -429,6 +435,13 @@ void checkForTimeout(TaskDef taskType, Task task) {
 		
 		Task task = null;
 		Map<String, Object> input = getTaskInput(taskToSchedule.getInputParameters(), workflow, null, null);
+
+		if ( input.get("timeoutSecondsOverride") != null){
```

Comment:
I do not think you need this entire block at all. The important task was to add to `WorkflowTask` and that task will be auto-unwrapped from JSON to WorkflowTask instance

Reply:
Besides that, having additional parameters in `taskToSchedule.getInputParameters()` wight impact matching logic. See how WAIT matching logic works.

This generic solution will work but it need to be careful at which WAIT task this `timeoutSecondsOverride` can be applied.

Even if you need this, remove it from `input` once you read the data from it. In that case you won't break any matching logic.

Reply:
Updated the code to remove the **timeoutSecondsOverride** key/value from the input parameters